tools/p4mlir-translate/translate.cpp

```diff
@@ -1139,21 +1224,20 @@ bool P4HIRConverter::preorder(const P4::IR::MethodCallExpression *mce) {
             // TODO: Are there cases when we do not have l-value here?
             auto ref = resolveReference(bCall->appliedTo);
             auto loc = getLoc(builder, mce);
-            auto validityBitRef =
-                b.create<P4HIR::StructExtractRefOp>(loc, ref, P4HIR::HeaderType::validityBit);
-            if (bCall->name == P4::IR::Type_Header::setValid ||
-                bCall->name == P4::IR::Type_Header::setInvalid) {
-                auto validVal =
-                    getValidHeaderConstant(loc, bCall->name == P4::IR::Type_Header::setInvalid
-                                                    ? P4HIR::ValidityBit::Invalid
-                                                    : P4HIR::ValidityBit::Valid);
-                b.create<P4HIR::AssignOp>(loc, validVal, validityBitRef);
-            } else if (bCall->name == P4::IR::Type_Header::isValid) {
-                auto validityBit = b.create<P4HIR::ReadOp>(loc, validityBitRef);
-                callResult = builder.create<P4HIR::CmpOp>(loc, P4HIR::CmpOpKind::Eq, validityBit,
-                                                          getValidHeaderConstant(loc));
-            } else
-                BUG("Unsupported builtin method: %1%", bCall->name);
+
+            // Check if this is a method call on a header or header union
+            if (mce->method->is<P4::IR::Member>()) {
+                const auto *member = mce->method->to<P4::IR::Member>();
```

Comment:
you can just do `if (const auto *member = mce->method->to<P4::IR::Member>()) { ... }`